tests/src/string/src/main.sw

```diff
@@ -39,87 +39,90 @@ abi StringTest {
 
 impl StringTest for Contract {
     fn test_append() {
-        // Uncomment when https://github.com/FuelLabs/sway/issues/4408 is resolved.
-        // let mut string1 = String::new();
-        // let mut string2 = String::new();
-        // string1.push(NUMBER0);
-        // string1.push(NUMBER1);
-        // string1.push(NUMBER2);
-        // string2.push(NUMBER3);
-        // string2.push(NUMBER4);
-        // string2.push(NUMBER5);
-        // string1.append(string2);
-        // assert(string2.len() == 0);
-        // assert(string1.len() == 6);
-        // assert(string1.nth(0).unwrap() == NUMBER0);
-        // assert(string1.nth(1).unwrap() == NUMBER1);
-        // assert(string1.nth(2).unwrap() == NUMBER2);
-        // assert(string1.nth(3).unwrap() == NUMBER3);
-        // assert(string1.nth(4).unwrap() == NUMBER4);
-        // assert(string1.nth(5).unwrap() == NUMBER5);
-}
+        let mut string1 = String::new();
+        let mut string2 = String::new();
+
+        string1.push(NUMBER0);
+        string1.push(NUMBER1);
+        string1.push(NUMBER2);
+        string2.push(NUMBER3);
+        string2.push(NUMBER4);
+        string2.push(NUMBER5);
+
+        assert(string2.len() == 3);
+        assert(string1.len() == 3);
+
+        string1.append(string2);
+
+        assert(string2.len() == 0);
```

Comment:
Should `string2` give away its values or should `string1` only have those values added?

Reply:
Yes https://github.com/FuelLabs/sway/blob/bbe746dafa102dba3ca4a392f23548515e122343/sway-lib-std/src/bytes.sw#L640